src/lib/y2partitioner/dialogs/blk_device_resize.rb

```diff
@@ -148,6 +162,105 @@ def used_size_info
+      # system and it is mounted)
+      #
+      # @return [Boolean]
+      def try_unmount?
+        controller.committed_current_filesystem? &&
+          controller.mounted_committed_filesystem?
+      end
+
+      # Tries to unmount when shrinking the device
+      #
+      # @return [Boolean] true if the device supports mounted shrinking or the device was
+      #   correctly unmounted or user decides to continue; false if user cancels.
+      def try_unmount_for_shrinking
+        return true unless shrinking? && controller.need_unmount_for_shrinking?
+
+        # TRANSLATORS: Note added to the dialog for trying to unmount a device
+        note = _("It is not possible to shrink the file system while it is mounted.")
+
+        immediate_unmount(controller.committed_device, note: note)
+      end
+
+      # Tries to unmount when growing the device
+      #
+      # @return [Boolean] true if the device supports mounted growing or the device was
+      #   correctly unmounted or user decides to continue; false if user cancels.
+      def try_unmount_for_growing
+        return true unless growing? && controller.need_unmount_for_growing?
+
+        # TRANSLATORS: Note added to the dialog for trying to unmount a device
+        note = _("It is not possible to extend the file system while it is mounted.")
+
+        immediate_unmount(controller.committed_device, note: note)
+      end
+
+      # Tries to unmount when performing big growing
+      #
+      # @return [Boolean] true if the device was correctly unmounted or user decides to
+      #   continue; false if user cancels.
+      def try_unmount_for_big_growing
+        return true unless big_growing? && controller.mounted_committed_filesystem?
+
+        message = format(
+          # TRANSLATORS: Text for the dialog for trying to unmount a device. It is used
+          # when a device is tried to be resized by growing it more than 1 GiB.
+          _("You are extending a mounted filesystem by %{gibs} Gigabyte.\n" \
+            "This may be quite slow and can take hours. You might possibly want \n" \
+            "to consider umounting the filesystem, which will increase speed of \n" \
+            "resize task a lot.\n\n" \
+            "You can try to unmount it now, continue without unmounting or cancel.\n" \
+            "Click Cancel unless you know exactly what you are doing."),
+          gibs: growing_size.to_i / Y2Storage::DiskSize.GiB(1).to_i
+        )
+
+        immediate_unmount(controller.committed_device, full_message: message, allow_continue: true)
+      end
+
+      # Whether the device is going to be shrinked
+      #
+      # @return [Boolean]
+      def shrinking?
+        selected_size < device.size
+      end
+
+      # Whether the device is going to be growed
+      #
+      # @return [Boolean]
+      def growing?
+        selected_size > device.size
+      end
+
+      # Whether the device is going to be growed more than 1 GiB
```

Comment:
Can you please add link why more then 1 GiB is big growing? So we can in future check if value is still reasonable.